### DISCUSSION TITLE
APPLE MPS Support

### DISCUSSION THREAD

**Discussion by aminekhelif** [pypose/pypose#336] [Q&A]
Hi,

I've been exploring the PyPose library for some IMU training tasks and I'm particularly interested in leveraging the Apple Metal Performance Shaders (MPS) backend for its potential performance benefits on Apple hardware. However, I've encountered several errors during my initial attempts, indicating that some features might not be supported with the MPS backend.

Could anyone provide clarity on whether PyPose officially supports the Apple MPS backend? If so, are there any specific configurations or versions of the library that I need to be aware of to utilize MPS effectively? 

Thank you

**Comment by zitongzhan** [comment]
We haven't been officially supporting MPS, as not all pytorch ops are fully implemented in MPS. However, it is something good to have in the future.
